app/src/main/java/com/example/drawingactivity/DialogUtils.kt

```diff
@@ -137,6 +136,183 @@ object DialogUtils {
             }
         }
     }
+
+    fun colorPickerBottomSheet(
+        context: Context,
+        previousColor: FloatArray,
+        /**
+         * Lambda that provides the string of the current color.
+         * Returns nothing. Used to save the hex value of the current color.
+         */
+        onSubmit: ((FloatArray?, String?) -> Unit)
+    ) {
+        val sheet = BottomSheetDialog(context)
+        val view = LayoutInflater.from(context).inflate(R.layout.color_dialog, null)
+        val huePicker = view.findViewById<HueView>(R.id.huePicker)
+        val gradientPicker = view.findViewById<GradientView>(R.id.gradientPicker)
+        val hexText = view.findViewById<ForcableEditText>(R.id.hexText)
```

Comment:
in the case that there are multiple things with hexcode for text, may want to specify for the name, in this case something like colorHex or hexColor. That doesn't apply here however, just food for thought.

Reply:
I'll try to keep that in mind for the future. My Text names could be better in the case where this does need to be scaled further